skdecide/builders/domain/scheduling/scheduling_domains_modelling.py

```diff
@@ -4,13 +4,72 @@
 
 from __future__ import annotations
 
-from copy import deepcopy
+from collections.abc import Collection
+from copy import copy, deepcopy
 from enum import Enum
-from typing import Dict, List, Optional, Set, Union
+from typing import Dict, Iterable, Optional, Set, Tuple, Union
 
 from skdecide.builders.domain.scheduling.task import Task
 
 
+def rebuild_tasks_complete_details_dict(state: State) -> Dict[int, Task]:
```

Comment:
do this "rebuild_***" function deserve some docstring ? I think their name + signature are fine, but you may think differently